components/script/dom/blob.rs

```diff
@@ -239,7 +239,8 @@ impl Blob {
         };
 
         let (tx, rx) = ipc::channel(self.global().time_profiler_chan().clone()).unwrap();
-        let msg = FileManagerThreadMsg::PromoteMemory(blob_buf, set_valid, tx, origin.clone());
+        let id = Uuid::new_v4();
+        let msg = FileManagerThreadMsg::PromoteMemory(id, blob_buf, set_valid, tx, origin.clone());
         self.send_to_file_manager(msg);
 
         match rx.recv().unwrap() {
```

Comment:
Based on the issue description, I think we want to remove this `rx.recv()` but I wasn't 100% sure.

If so, would we just assign `*self.blob_impl.borrow_mut()` and return `id`?